loopchain/blockchain/blocks/v0_3/block.py

```diff
@@ -25,6 +27,26 @@ class BlockHeader(BaseBlockHeader):
     def complained(self):
         return self.leader_votes_hash != Hash32.empty()
 
+    @property
+    def prep_changed(self) -> Optional[PrepChangedReason]:
+        """Return reason for prep changed
+
+        :return: None means there is no change.
+        """
+        if self.next_reps_hash == Hash32.empty():
+            return None
+
+        if self.next_leader == ExternalAddress.empty():
+            return PrepChangedReason.TERM_END
+
+        return PrepChangedReason.PENALTY
+
+    @property
+    def revealed_next_reps_hash(self):
+        if self.prep_changed:
+            return self.next_reps_hash
+        return self.reps_hash
+
```

Comment:
those three method became used at both v0.1 and v0.3.
How about extract those things at `BaseBlockHeader`?

Reply:
good idea!

Reply:
But it's not simple. Because BaseBlockHeader is set to the data class. I postpone this to the next one.